core/src/test/java/io/confluent/rest/SslTest.java

```diff
@@ -0,0 +1,327 @@
+    props.put(RestConfig.REST_PROTOCOL_CONFIG, RestConfig.REST_PROTOCOL_HTTP);
+    TestRestConfig config = new TestRestConfig(props);
+    SslTestApplication app = new SslTestApplication(config);
+    try {
+      app.start();
+
+      makeGetRequestAndVerify200("http://localhost:" + RestConfig.PORT_CONFIG_DEFAULT + "/test");
+
+      try {
+        makeGetRequestAndVerify200("https://localhost:" + (RestConfig.PORT_HTTPS_CONFIG_DEFAULT) + "/test",
+                clientKeystore.getAbsolutePath(), SSL_PASSWORD, SSL_PASSWORD);
+      } catch (HttpHostConnectException hhce) {
+        // expected.
+      }
+    } finally {
+      app.stop();
+    }
+  }
+
+  @Test
+  public void testHttpsWithNoClientCertAndNoServerTruststore() throws Exception {
+    Properties props = new Properties();
+    props.put(RestConfig.REST_PROTOCOL_CONFIG, RestConfig.REST_PROTOCOL_HTTPS);
+    configServerKeystore(props);
+    TestRestConfig config = new TestRestConfig(props);
+    SslTestApplication app = new SslTestApplication(config);
+    try {
+      app.start();
+
+      makeGetRequestAndVerify200("https://localhost:" + (RestConfig.PORT_HTTPS_CONFIG_DEFAULT) + "/test");
+    } finally {
+      app.stop();
+    }
+  }
+
+  @Test
+  public void testHttpsWithAuthAndBadClientCert() throws Exception {
+    Properties props = new Properties();
+    props.put(RestConfig.REST_PROTOCOL_CONFIG, RestConfig.REST_PROTOCOL_HTTPS);
+    configServerKeystore(props);
+    configServerTruststore(props);
+    enableSslClientAuth(props);
+    TestRestConfig config = new TestRestConfig(props);
+    SslTestApplication app = new SslTestApplication(config);
+    try {
+      app.start();
+
+      // create a new client cert that isn't in the server's trust store.
+      File untrustedClient = File.createTempFile("SslTest-client-keystore", ".jks");
+      Map<String, X509Certificate> certs = new HashMap<>();
+      createKeystoreWithCert(untrustedClient, "client", certs);
+
+      try {
+        makeGetRequestAndVerify200("https://localhost:" + (RestConfig.PORT_HTTPS_CONFIG_DEFAULT) + "/test",
+                untrustedClient.getAbsolutePath(), SSL_PASSWORD, SSL_PASSWORD);
+      } catch (SSLHandshakeException she) {
+        // expected.
+      } catch (SocketException se) {
+        // expected. SocketException is thrown when SSLHandshakeException is caught.
+      }
+    } finally {
+      app.stop();
+    }
+  }
+
+  @Test
+  public void testHttpsWithAuthAndNoClientCert() throws Exception {
+    Properties props = new Properties();
+    props.put(RestConfig.REST_PROTOCOL_CONFIG, RestConfig.REST_PROTOCOL_HTTPS);
+    configServerKeystore(props);
+    configServerTruststore(props);
+    enableSslClientAuth(props);
+    TestRestConfig config = new TestRestConfig(props);
+    SslTestApplication app = new SslTestApplication(config);
+    try {
+      app.start();
+
+      try {
+        makeGetRequestAndVerify200("https://localhost:" + (RestConfig.PORT_HTTPS_CONFIG_DEFAULT) + "/test");
```

Comment:
I realize you're just reusing code, but these calls might be confusing in tests that are expected to fail -- the name implies you'll check the 200 result, but in fact you won't since it throws an exception first, right?